diff --git a/docroot/sites/all/modules/contrib/addressfield_tokens/addressfield_tokens.components.inc b/docroot/sites/all/modules/contrib/addressfield_tokens/addressfield_tokens.components.inc
index 840bd5038..a516dee7a 100644
--- a/docroot/sites/all/modules/contrib/addressfield_tokens/addressfield_tokens.components.inc
+++ b/docroot/sites/all/modules/contrib/addressfield_tokens/addressfield_tokens.components.inc
@@ -112,7 +112,7 @@ function _webform_render_addressfield($component, $value = NULL, $filter = TRUE)
     '#attributes'       => $component['extra']['attributes'],
     '#theme_wrappers'   => array('webform_element'),
     '#description'      => $filter ? _webform_filter_descriptions($component['extra']['description']) : $component['extra']['description'],
-    '#required'         => $component['mandatory'],
+    '#required'         => $component['required'],
     '#weight'           => $component['weight'],
     '#translatable'     => array(
       'title',
@@ -164,7 +164,7 @@ function _webform_render_addressfield($component, $value = NULL, $filter = TRUE)
       // Hide the country element only if there is one option and the whole field
       // is required, otherwise there will always be an additional None option.
       // @see addressfield_format_address_hide_country()
-      if (!empty($handlers['address-hide-country']) && count($element['country']['#options']) == 1 && $component['mandatory']) {
+      if (!empty($handlers['address-hide-country']) && count($element['country']['#options']) == 1 && $component['required']) {
         $element['country']['#access'] =  FALSE;
       }
     }
