diff -r 3797cd3da1c2 administerusersbyrole.module
--- a/administerusersbyrole.module	Tue Sep 04 13:48:06 2012 +0200
+++ b/administerusersbyrole.module	Tue Sep 04 14:03:44 2012 +0200
@@ -94,13 +94,20 @@

   $allow = TRUE;
-  foreach ($account->roles as $rid => $role) {
+  foreach ($account->roles as $rid => $role_name) {
+    if(is_numeric($role_name)){
+      $role = user_role_load($role_name);
+      if(!isset($role->name)){
+        continue;
+      }
+      $role_name = $role->name;
+    }
     if ($rid === DRUPAL_AUTHENTICATED_RID) {
       continue;
     }
-    if (user_access(_administerusersbyrole_build_perm_string($role, 'edit', TRUE))) {
+    if (user_access(_administerusersbyrole_build_perm_string($role_name, 'edit', TRUE))) {
       return TRUE;
     }
-    if (!user_access(_administerusersbyrole_build_perm_string($role, 'edit', FALSE))) {
+    if (!user_access(_administerusersbyrole_build_perm_string($role_name, 'edit', FALSE))) {
       $allow = FALSE;
     }
   }
