diff --git a/docroot/sites/all/modules/contrib/google_analytics/googleanalytics.module b/docroot/sites/all/modules/contrib/google_analytics/googleanalytics.module
index 918135f..227e0f0 100644
--- a/docroot/sites/all/modules/contrib/google_analytics/googleanalytics.module
+++ b/docroot/sites/all/modules/contrib/google_analytics/googleanalytics.module
@@ -79,6 +79,10 @@ function googleanalytics_page_alter(&$page) {
   global $user;
 
   $id = variable_get('googleanalytics_account', '');
+  // Extended to use pathwise Google analytic.
+  if (function_exists('ismaili_custom_extend_google_anaytics')) {
+    ismaili_custom_extend_google_anaytics($id);
+  }
 
   // Get page status code for visibility filtering.
   $status = drupal_get_http_header('Status');
