diff --git a/docroot/sites/ismaili/modules/contrib/nat/nat.module b/docroot/sites/ismaili/modules/contrib/nat/nat.module
index 46cc5fa..7e8efe2 100644
--- a/docroot/sites/ismaili/modules/contrib/nat/nat.module
+++ b/docroot/sites/ismaili/modules/contrib/nat/nat.module
@@ -615,8 +615,21 @@ function _nat_update_terms($node) {
  *   The fields to synchronize: an associative array of term=>node fields.
  */
 function _nat_save_term($node, $hierarchy, $term, $entities, $fields) {
-  $term->parent = isset($hierarchy[$term->vid]) ? $hierarchy[$term->vid] : array(0);
-
+  if (isset($hierarchy[$term->vid])) {
+    $term->parent = $hierarchy[$term->vid];
+  }
+  else {
+    // Find term parents
+    $parents = taxonomy_get_parents($term->tid);
+    if(count($parents)) {
+      // Preserves term's parent if it exists
+      $parent = current($parents);
+      $term->parent = array($parent->tid => $parent->tid);
+    }
+    else {
+      $term->parent = array();
+    }
+  }
   foreach ($fields as $term_field => $node_field) {
     $resolved_fields = _nat_field_associate($node, $entities, $term_field, $node_field);
     foreach ($resolved_fields as $term_field_resolved => $node_field_resolved) {
