Index: plupload.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/plupload/plupload.module,v
retrieving revision 1.12
diff -u -p -r1.12 plupload.module
--- plupload.module	16 Sep 2010 17:40:31 -0000	1.12
+++ plupload.module	1 Oct 2010 05:44:09 -0000
@@ -249,9 +249,11 @@ function plupload_handle_uploads() {
   $file_name = isset($_REQUEST["name"]) ? $_REQUEST["name"] : '';
 
   // Check the file name for security reasons
+  /*
   if (empty($file_name) || !preg_match('/^[\w\._]+\.tmp$/', $file_name)) {
     die('{"jsonrpc" : "2.0", "error" : {"code": 105, "message": "Invalid temporary file name."}, "id" : "id"}');
   }
+  */
   
   // Look for the content type header
   if (isset($_SERVER["HTTP_CONTENT_TYPE"])) {
