diff --git a/filedepot.module b/filedepot.module
index b8a2a87..19aba33 100755
--- a/filedepot.module
+++ b/filedepot.module
@@ -1254,12 +1254,7 @@ function filedepot_download($node, $fid, $version, $mode = '') {
         ));
         return drupal_access_denied();
       }
-      if ($mode == 'moderator') {
-        $filepath = $filedepot->root_storage_path . "{$cid}/submissions/{$tempname}";
-      }
-      else {
-        $filepath = $filedepot->root_storage_path . "{$cid}/{$fname}";
-      }
+      $filepath = 'private://' . $fname;
 
       list($scheme, $target) = explode('://', $filepath, 2);
       if ($filedepot->checkPermission($cid, 'view') === FALSE) {
