diff --git a/filedepot.module b/filedepot.module
index b8a2a87..d9c80b5 100755
--- a/filedepot.module
+++ b/filedepot.module
@@ -1166,7 +1166,7 @@ function filedepot_download($node, $fid, $version, $mode = '') {
         $type = mime_header_encode($file->filemime);
         // By default, serve images, text, and flash content for display rather than
         // download. Or if variable 'filefield_inline_types' is set, use its patterns.
-        $inline_types = variable_get('filefield_inline_types', array('^text/', '^image/', 'flash$'));
+        $inline_types = array();
         $disposition = 'attachment';
         foreach ($inline_types as $inline_type) {
           // Exclamation marks are used as delimiters to avoid escaping slashes.
@@ -1561,7 +1561,7 @@ function filedepot_file_download() {
       $disposition = 'attachment';
       // By default, serve images, text, and flash content for display rather than
       // download. Or if variable 'filefield_inline_types' is set, use its patterns.
-      $inline_types = variable_get('filefield_inline_types', array('^text/', '^image/', 'flash$'));
+      $inline_types = array();
       $disposition = 'attachment';
       foreach ($inline_types as $inline_type) {
         // Exclamation marks are used as delimiters to avoid escaping slashes.
@@ -1634,7 +1634,7 @@ function filedepot_file_download() {
         $disposition = 'attachment';
         // By default, serve images, text, and flash content for display rather than
         // download. Or if variable 'filefield_inline_types' is set, use its patterns.
-        $inline_types = variable_get('filefield_inline_types', array('^text/', '^image/', 'flash$'));
+        $inline_types = array();
         $disposition = 'attachment';
         foreach ($inline_types as $inline_type) {
           // Exclamation marks are used as delimiters to avoid escaping slashes.
