diff --git a/ctools.module b/ctools.module
index 778a220..1ceac59 100644
--- a/ctools.module
+++ b/ctools.module
@@ -758,15 +758,17 @@ function ctools_process(&$variables, $hook) {
     $variables['classes_array'] = array_diff($variables['classes_array'], $remove_classes);
   }
 
+   $classes_array = $variables['classes_array'];
   // Update the classes within the attributes array to match the classes array
   if (isset($variables['attributes_array']['class'])) {
-    $variables['attributes_array']['class'] = $variables['classes_array'];
+    $variables['attributes_array']['class'] = array_unique(array_merge($variables['attributes_array']['class'], $classes_array));
     $variables['attributes'] = $variables['attributes_array'] ? drupal_attributes($variables['attributes_array']) : '';
+    $classes_array = array_merge($classes_array, $variables['attributes_array']['class']);
   }
 
   // Since this runs after template_process(), we need to re-implode the
   // classes array.
-  $variables['classes'] = implode(' ', $variables['classes_array']);
+  $variables['classes'] = implode(' ', array_unique($classes_array));
 }
 
 // -----------------------------------------------------------------------
