diff --git a/docroot/sites/akdn/modules/contrib/galleryformatter/theme/galleryformatter.js b/docroot/sites/akdn/modules/contrib/galleryformatter/theme/galleryformatter.js
index 39ecc50..b52e750 100644
--- a/docroot/sites/akdn/modules/contrib/galleryformatter/theme/galleryformatter.js
+++ b/docroot/sites/akdn/modules/contrib/galleryformatter/theme/galleryformatter.js
@@ -39,7 +39,7 @@
      * Only start the thumbs carrousel if needed
      */
     if (($thumbsLi.size() * liWidth) > $thumbs.width()) {
-      $('ul', $thumbs).width('99999px');
+      $('ul', $thumbs).width($thumbsLi.size() * liWidth + 'px');
       $thumbs.infiniteCarousel();
       $thumbsLi = $('li', $thumbs); // we need to reselect because infiniteCarousel inserts new empty li elements if necessary
       // we need to reselect because infiniteCarousel inserts new empty li elements if necessary
