--- a/nat.module	2018-06-20 12:29:56.238352000 +0530
+++ b/nat.module	2018-06-20 12:32:20.794901267 +0530
@@ -330,6 +330,14 @@
     WHERE n.nid = :nid", array(':nid' => $nid)
   );
   foreach ($result as $term) {
+    if (module_exists('entity_translation')) {
+      global $language;
+      $term_load = taxonomy_term_load($term->tid);
+      if(array_key_exists('name_field', $term_load)){
+        $term->name = array_key_exists($language->language, $term_load->name_field)?$term_load->name_field[$language->language][0]['safe_value']:$term_load->name_field['en'][0]['safe_value'];
+        $term->language = $language->language;
+      }
+    }
     $return[$term->tid] = $term;
   }
 
@@ -623,6 +631,18 @@
       $term->$term_field_resolved = $node_field_resolved;
     }
   }
+  if (module_exists('entity_translation') && isset($term->tid) && $term->tid) {
+    global $language;
+    $translation = array(
+        'translate' => 0,
+        'status' => 1,
+        'language' => $language->language, 
+        'source' => $term->translations->original, 
+    );
+    $handler = entity_translation_get_handler('taxonomy_term', $term, TRUE);
+    $values['name_field'][$language->language]['0']['value'] =   $term->name;
+    $handler->setTranslation($translation, $values);
+  }
 
   taxonomy_term_save($term);
 
